<a href="https://colab.research.google.com/github/YashM246/Hamoye-Internship-Fall-2022/blob/main/Hamoye_Python_for_ML.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Hamoye Python for Machine Learning Fall 2022 Internship

## Name            - Yash Malode<br><br>

## Importing required Libraries

In [71]:
import pandas as pd
import numpy as np
from google.colab import files
import io

## Importing csv file

In [ ]:
uploaded = files.upload()
df = pd.read_csv(io.BytesIO(uploaded['FoodBalanceSheets_E_Africa_NOFLAG.csv']), encoding="latin-1")
df.head()

Saving FoodBalanceSheets_E_Africa_NOFLAG.csv to FoodBalanceSheets_E_Africa_NOFLAG (1).csv


Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

## Question 1 - Number and Percentage of missing data in 2014 to 3 decimal places

In [ ]:
print("Total Number -",df['Y2014'].isnull().sum())

print("Percentage -",round(((df['Y2014'].isnull().sum())*100)/len(df),3))

Total Number - 1589
Percentage - 2.607


## Question 2 - Perform groupby on Element. Total sum of Processing in 2017

In [ ]:
df.groupby('Element')['Y2017'].sum()

Element
Domestic supply quantity                  2088198.10
Export Quantity                            182338.80
Fat supply quantity (g/capita/day)          10253.84
Feed                                       223705.68
Food                                      1258888.28
Food supply (kcal/capita/day)              454681.00
Food supply quantity (kg/capita/yr)         48690.04
Import Quantity                            294559.09
Losses                                     160614.00
Other uses (non-food)                       91645.97
Processing                                 292836.00
Production                                2030056.89
Protein supply quantity (g/capita/day)      11842.45
Residuals                                   35500.00
Seed                                        24870.14
Stock Variation                             54316.91
Total Population - Both sexes             1112641.00
Tourist consumption                            91.00
Name: Y2017, dtype: float64

## Question 5 - Highest Sum Area in 2017

In [ ]:
df.groupby('Area')['Y2017'].sum()

Area
Algeria                         325644.27
Angola                          229159.57
Benin                           124771.22
Botswana                         22101.30
Burkina Faso                    101855.07
Cabo Verde                       14650.74
Cameroon                        232030.43
Central African Republic         29937.00
Chad                             71594.68
Comoros                             59.84
Congo                            41181.68
Côte d'Ivoire                   224599.01
Djibouti                         22729.91
Egypt                           866379.92
Eswatini                         54343.33
Ethiopia                        448683.76
Ethiopia PDR                         0.00
Gabon                            27979.64
Gambia                           23154.18
Ghana                           337599.06
Guinea                           98138.87
Guinea-Bissau                    19102.77
Kenya                           264660.66
Lesotho                      

In [ ]:
df.groupby('Area')['Y2017'].sum().max()

1483268.23

Here we can see the max value is 1483268.23 which belongs to Nigeria<br><br>
## Question 6 - Area with 7th lowest Sum in 2017

In [ ]:
sorted(df.groupby('Area')['Y2017'].sum())

[0.0,
 0.0,
 59.839999999999996,
 442.34000000000003,
 12662.630000000001,
 14650.74,
 19102.77,
 21267.96,
 22101.3,
 22729.91,
 23154.18,
 27979.64,
 29342.2,
 29874.89,
 29937.0,
 41181.68,
 49841.88,
 51114.83,
 54343.33,
 55311.33,
 71594.68,
 73663.69,
 75919.34,
 95681.15,
 98138.87,
 101855.06999999999,
 103223.77,
 124167.2,
 124771.22,
 126707.58,
 131197.73,
 149928.33,
 156665.46,
 161407.98,
 181098.71,
 213950.38,
 224599.01,
 229159.57,
 232030.43,
 239931.92,
 264660.66,
 322616.85,
 325644.27,
 337599.06,
 388495.36,
 448683.76,
 517590.54,
 866379.92,
 1483268.23]

As we can see the 7th lowest value is 19102.77<br>
This value belongs to Guinea Bissau<br><br>
## Question 7 - Total Protein Supply in Madagascar in 2015

In [68]:
df2 = df.groupby(['Area','Element'], as_index=False)['Y2015'].sum()
df2.iloc[410:430,:]

Area                                 Element     Y2015
410     Liberia           Total Population - Both sexes   4472.00
411  Madagascar                Domestic supply quantity  31214.98
412  Madagascar                         Export Quantity    494.75
413  Madagascar      Fat supply quantity (g/capita/day)     91.85
414  Madagascar                                    Feed   2070.22
415  Madagascar                                    Food  21120.65
416  Madagascar           Food supply (kcal/capita/day)   7685.00
417  Madagascar     Food supply quantity (kg/capita/yr)    871.59
418  Madagascar                         Import Quantity   1721.80
419  Madagascar                                  Losses   2176.00
420  Madagascar                   Other uses (non-food)   3957.12
421  Madagascar                              Processing   1803.00
422  Madagascar                              Production  29482.89
423  Madagascar  Protein supply quantity (g/capita/day)    173.05
424  Madagascar                               Residuals   -616.00
425  Madagascar                                    Seed    699.00
426  Madagascar                         Stock Variation   -505.00
427  Madagascar           Total Population - Both sexes  24234.00
428      Malawi                Domestic supply quantity  43271.06
429      Malawi                         Export Quantity    583.02

## Question 10 - Least Correlation with Element Code

In [48]:
correlation = df.corr()
correlation

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

In [49]:
correlation.iloc[3:,2]

Y2014    0.024457
Y2015    0.023889
Y2016    0.023444
Y2017    0.024254
Y2018    0.024279
Name: Element Code, dtype: float64

As we can see Year 2016 has least correlation

## Question 11 - Unique Countries

In [54]:
df.groupby(['Area Code'], as_index=False)['Area'].count()

Area Code  Area
0           4  1313
1           7  1424
2          20  1378
3          32  1460
4          35  1260
5          37  1344
6          39  1343
7          45    45
8          46  1413
9          53  1285
10         59  1488
11         62    39
12         72  1095
13         74  1245
14         75  1168
15         81  1354
16         90  1410
17        107  1467
18        114  1560
19        122  1150
20        123  1206
21        129  1439
22        130  1447
23        133  1293
24        136  1195
25        137  1378
26        143  1393
27        144  1427
28        147  1283
29        158  1403
30        159  1474
31        175  1128
32        181  1393
33        184  1426
34        193  1177
35        195  1462
36        196    68
37        197  1242
38        202  1399
39        206    53
40        209  1284
41        215  1419
42        217  1294
43        222  1338
44        226  1458
45        233  1409
46        238  1447
47        251  1451
48        276  1316

0 to 48, therefore 49 unique countries

## Question 19 - Mean and SD for whole dataset for Year 2017 upto 2 decimal places

In [70]:
print("Mean - ",round(df['Y2017'].mean(),2))
print("SD - ",round(df['Y2017'].std(axis = 0),2))

Mean -  140.92
SD -  1671.86
